scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala

```diff
@@ -73,9 +73,21 @@ class Router(formatOps: FormatOps) {
     val newlines = formatToken.newlinesBetween
 
     formatToken match {
-      case FormatToken(_: T.BOF, _, _) =>
+      case FormatToken(_: T.BOF, right, _) =>
+        val policy = right match {
+          case T.Ident(name) // shebang in .sbt/.sc files
+              if name.startsWith("#!") &&
+                style.runner.dialect.allowToplevelTerms =>
```

Comment:
As far as I understand, it works for `runner.dialect = sbt`.

But shebangs much more commonly used in [ammonite](https://ammonite.io/#ScalaScripts) scripts. It may be makes sense to create alias [here](https://github.com/scalameta/scalafmt/blob/master/scalafmt-core/shared/src/main/scala/org/scalafmt/config/ScalafmtRunner.scala#L98)

```scala
val ScalaScripts = Sbt1
```

Reply:
the dialect for sbt is not set explicitly, the dynamic runner does it automatically: https://github.com/scalameta/scalafmt/blob/a7bb6920a8ba486cfbff74d056037dfeb36c7e33/scalafmt-dynamic/src/main/scala/org/scalafmt/dynamic/ScalafmtDynamic.scala#L199

problem is, as the code above shows (and that's just what we fixed in #2075 today), the dialect used for `.sc` files is not a given instance, it's based on the main dialect configured for scala, with `.forSbt` invoked (https://github.com/kitbellew/scalafmt/blob/8be8f7beb3194f2224b9580ec55e2bab909ba8e4/scalafmt-core/shared/src/main/scala/org/scalafmt/config/ScalafmtRunner.scala#L30)

so, in this method, i don't know the name of the file (to check for `.sc`) and i can't compare dialect to `sbt` or `sbt1` or `sbt0137`. so i check for `allowToplevelTerms` flag as the only thing remaining.

Reply:
ok, i figured out a way to pass the name of the file to this code. in order to be able to test, had to modify HasTests to take a filename to use and had to invoke `.forSbt` inside the formatter (hence changes to core and dynamic runners to stop doing that). PTAL.